File: clients/typescript-angular/API/Client/Margin.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    export interface Margin {

        account: number;

        currency: string;

        riskLimit?: number;

        prevState?: string;

        state?: string;

        action?: string;

        amount?: number;

        pendingCredit?: number;

        pendingDebit?: number;

        prevRealisedPnl?: number;

        prevUnrealisedPnl?: number;

        grossComm?: number;

        grossOpenCost?: number;

        grossOpenPremium?: number;

        grossExecCost?: number;

        grossMarkValue?: number;

        riskValue?: number;

        taxableMargin?: number;

        initMargin?: number;

        maintMargin?: number;

        sessionMargin?: number;

        targetExcessMargin?: number;

        varMargin?: number;

        realisedPnl?: number;

        unrealisedPnl?: number;

        indicativeTax?: number;

        unrealisedProfit?: number;

        walletBalance?: number;

        marginBalance?: number;

        marginBalancePcnt?: number;

        marginLeverage?: number;

        marginUsedPcnt?: number;

        excessMargin?: number;

        excessMarginPcnt?: number;

        availableMargin?: number;

        withdrawableMargin?: number;

        timestamp?: date;
    }

}

File: clients/scala/src/main/scala/io/swagger/client/model/Margin.scala
package io.swagger.client.model

import java.util.Date



case class Margin (
  account: Number,
  currency: String,
  riskLimit: Number,
  prevState: String,
  state: String,
  action: String,
  amount: Number,
  pendingCredit: Number,
  pendingDebit: Number,
  confirmedDebit: Number,
  prevRealisedPnl: Number,
  prevUnrealisedPnl: Number,
  grossComm: Number,
  grossOpenCost: Number,
  grossOpenPremium: Number,
  grossExecCost: Number,
  grossMarkValue: Number,
  riskValue: Number,
  taxableMargin: Number,
  initMargin: Number,
  maintMargin: Number,
  sessionMargin: Number,
  targetExcessMargin: Number,
  varMargin: Number,
  realisedPnl: Number,
  unrealisedPnl: Number,
  indicativeTax: Number,
  unrealisedProfit: Number,
  syntheticMargin: Number,
  walletBalance: Number,
  marginBalance: Number,
  marginBalancePcnt: Double,
  marginLeverage: Double,
  marginUsedPcnt: Double,
  excessMargin: Number,
  excessMarginPcnt: Double,
  availableMargin: Number,
  withdrawableMargin: Number,
  timestamp: Date,
  grossLastValue: Number,
  commission: Double)
  
